pkg/dynamiclistener/server.go

```diff
@@ -173,6 +162,31 @@ func (s *Server) updateIPs(savedIPs map[string]bool) map[string]bool {
 	return allIPs
 }
 
+func (s *Server) reconcileGeneratedCert(cfg *v3.ListenConfig) error {
```

Comment:
This is done only as a backup, in case the original save fails. If we want to simplify we can just do the save on generation below.